compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -307,7 +307,7 @@ impl<'tcx> HirTyLowerer<'tcx> for FnCtxt<'_, 'tcx> {
             ty::Alias(ty::Projection | ty::Inherent | ty::Weak, _)
                 if !ty.has_escaping_bound_vars() =>
```

Comment:
this is scuffed lol

Reply:
yes :3, we could try to use `with_replaced_escaping_bound_vars`, but also, I dislike the fact that we're normalizing here in the first place :<

Reply:
well, we need to normalize here so we can determine the adt. but yeah, i'm gonna leave it.